docs/design/disaster-recovery-optimization.md

```diff
@@ -0,0 +1,52 @@
+# FEATURE: Regional Disaster Recovery Optimization
+
+## Summary
+
+Implement use of `bluestore-rdr` as the OSD backend store for customers using ODF with Regional Disaster Recovery (RDR).
+
+## Motivation
+
+- Ceph uses BlueStore as a special-purpose storage back end designed specifically for managing data on disk for Ceph OSD workloads. Ceph team is working on improving this backend storage that will help in significantly improving the performance, especially in disaster recovery scenarios. This new OSD backend is known as bluestore-rdr. In the future, other backends will also need to be supported, such as seastore.
+
+## Proposal
+
+- Enable the optimized OSD backend store for RDR clusters.
+- Enable this optimization for both new (Greenfield) and existing (brownfield) customers.
+
+### Greenfield
+- If `ocs.openshift.io/optimizeDisasterRecovery: true` annotation is available in the `StorageCluster` then configure OSD store type in the `cephCluster` as:
+
+```YAML
+spec:
+  Storage:
+    store:
+        type: bluestore-rdr
+```
+
+### Brownfield
+- An existing ODF cluster can be updated to use the new OSD backend store.
+- If `spec.mirroring.enabled` is set to `true` and ODF version is greater than equal to 4.14, then configure OSD store in the `cephCluster` as:
+  
+```YAML
+spec:
+  Storage:
+    store:
+        type: bluestore-rdr
+        updateStore: yes-really-update-store
+```
+
+## API Changes
+- `storageCluster` status should indicate if the optimization has been applied successfully. 
+- This is useful for brownfield scenarios where the customers will migrate the OSDs to use the `bluestore-rdr` backend store
+
+```YAML
+status:
+    disasterRecoveryOptimization:
```

Comment:
Why do we need special status for the OSDs? As long as the CephCluster CR stays pending until conversion is done, then the storageCluster CR status would also show as not complete.

Reply:
This would be useful in case of brownfield cases. The UI could use the status to display some sort of message to the users that the cluster is undergoing optimization (OSD migration).  

This status will be generated from `cephCluster` status. So we can still skip adding this status in the `StorageCluster` and UI can directly query `cephCluster` to get the required information it needs.

Reply:
My initial thought about this special status was that, it can be a list of optimizations applied to the cluster. Since it is mostly going to be done behind the scenes, this could provide information on what has and has not been done on the cluster. This can even be surfaced up to multicluster UI.

Maybe something like
```
status:
    disasterRecoveryOptimizations:
    - OSDCreatedWithNewStore | OSDMigratedToNewStore
    - SomeOtherFutureOptimization
```

I agree that detailed status can be referenced from CephCluster. No point duplicating it.